neonvm/runner/main.go

```diff
@@ -480,6 +480,39 @@ func checkDevTun() bool {
 	return mode&os.ModeCharDevice == os.ModeCharDevice
 }
 
+func runInitScript(script string, logger *zap.Logger) error {
+	if len(script) == 0 {
+		return nil
+	}
+
+	// creates a tmp file with the script content
+	tmpFile, err := os.CreateTemp(os.TempDir(), "init-script-")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpFile.Name()) // clean up
+
+	if _, err := tmpFile.Write([]byte(script)); err != nil {
+		return err
+	}
+
+	if err := tmpFile.Close(); err != nil {
+		return err
+	}
+
+	if err := os.Chmod(tmpFile.Name(), 0755); err != nil {
+		return err
+	}
```

Comment:
out of curiosity, why the chmod here?

Reply:
That's `chmod +x`, to run the script

Reply:
Hm, I thought we didn't need that if we're not directly executing it (instead we're `/bin/sh <script name>`)

Reply:
Ok, I guess I did it out of habit.

Deleted it.